Mathlib/Algebra/Lie/Derivation/AdjointAction.lean

```diff
@@ -0,0 +1,138 @@
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Marbach
+-/
+import Mathlib.Algebra.Lie.Abelian
+import Mathlib.Algebra.Lie.Derivation.Basic
+import Mathlib.Algebra.Lie.OfAssociative
+
+/-!
+# Adjoint action of a Lie algebra on itself
+
+This file defines the *adjoint action* of a Lie algebra on itself, and establishes basic properties.
+
+## Main definitions
+
+- `LieDerivation.ad` : The adjoint action of a Lie algebra `L` on itself, seen as an homomorphism
+of Lie algebras from `L` to the Lie algebra of its derivations. The adjoint action is also defined
+in the `Mathlib.Algebra.Lie.OfAssociative.lean` file, under the name `LieAlgebra.ad`, as the
+homomorphism with values in the endormophisms of `L`.
+
+## Main statements
+
+- `LieDerivation.ad_eq_ad`: when seen as endomorphisms, both definitions coincide,
+- `LieDerivation.ad_ker_eq_center`: the kernel of the adjoint action is the center of `L`,
+- `LieDerivation.lie_der_ad_eq_ad_der`: the commutator of a derivation `D` and `ad(x)` is `ad(D x)`,
+- `LieDerivation.imageAd`: the image of the adjoint action is an ideal of the derivations.
+-/
+
+namespace LieDerivation
+
+section Inner
+
+variable (R L M : Type*) [CommRing R] [LieRing L] [LieAlgebra R L]
+    [AddCommGroup M] [Module R M] [LieRingModule L M] [LieModule R L M]
+
+/-- The inner derivation from the Lie algebra `L` to the Lie module `M`, associated with the right
+action of an element `m : M`. -/
+def inner (m : M) : LieDerivation R L M where
+  __ := (LieModule.toEndomorphism R L M : L →ₗ[R] Module.End R M).flip m
+  leibniz' := by
+    simp only [LinearMap.flip_apply, LieHom.coe_toLinearMap, LieHom.map_lie, LieHom.lie_apply,
+      LieModule.toEndomorphism_apply_apply, Module.End.lie_apply, forall_const]
+
+variable {R L M}
+
+lemma inner_apply (m : M) (a : L) : inner R L M m a = ⁅a, m⁆ := by
+  simp only [inner, LieDerivation.mk_coe, LinearMap.flip_apply, LieHom.coe_toLinearMap,
+    LieModule.toEndomorphism_apply_apply]
+
+end Inner
+
+section AdjointAction
+
+variable (R L : Type*) [CommRing R] [LieRing L] [LieAlgebra R L]
+
+/-- The adjoint action of a Lie algebra `L` on itself, seen as an homomorphism of Lie algebras from
+`L` to the Lie algebra of its derivations. This corresponds to the left action of `L` on itself, ie.
+such that, for `x y : L`, `ad(x) y = ⁅x, y⁆`. See also `LieAlgebra.ad` for the homomorphism with
+values in the endormophisms of `L`. -/
+def ad : L →ₗ⁅R⁆ LieDerivation R L L where
+  toFun := fun m ↦ inner R L L (- m)
+  map_add' := by
+    intro x y; ext a;
+    simp only [neg_add_rev, add_apply, inner_apply, lie_add]
+    abel
+  map_smul' := by
+    intro r x; ext a;
+    simp only [smul_apply, inner_apply, RingHom.id_apply, smul_neg, lie_smul, lie_neg]
+  map_lie' := by
+    intro x y; ext a;
+    simp only [commutator_apply, inner_apply, lie_neg]
+    rw [leibniz_lie, neg_lie, neg_lie, ← lie_skew x]
+    abel
```

Comment:
I don't think I can place the minus sign outside, since then it looks like I have `ad [x,y] = - [ad(x), ad(y)]`? In fact I added the minus sign in the first place to have a morphism of Lie algebras.

Reply:
Oh right, thanks; good point!

This makes me vaguely wonder if we should redefine `inner` to be what is currently `- inner`. Anyway, what you have is best for now I think.